### DISCUSSION TITLE
Multi display support

### DISCUSSION THREAD

**Discussion by TheKeyblader** [stride3d/stride#1305] [Q&A] [answered]
Hello 👋

Stride does support rendering inside multiple windows with same GameContext to allow sharing texture, etc..? 

If yes, can Stride render window in multiple threads ? 

If not, there is a way to implement without modifying stride source code ?

**Comment by tebjan** [comment, marked as answer]
hey, you can look into our implementation, it is without changing the stride source code.

what we implemented:
* multiple scene systems per game
* every scene system can have multiple windows and/or textures it renders into
* each window or texture can have its own graphics compositor and postfx
* multiple games per application, which have all of the above
* shared textures

some code entry points:
https://github.com/vvvv/VL.Stride/blob/preview/2021.4/packages/VL.Stride.Runtime/src/Games/GameWindowRenderer.cs
https://github.com/vvvv/VL.Stride/blob/preview/2021.4/packages/VL.Stride.Runtime/src/Engine/SceneInstanceRenderer.cs
https://github.com/vvvv/VL.Stride/blob/preview/2021.4/packages/VL.Stride.Runtime/src/Engine/SceneInstanceSystem.cs
and this renders multiple scene renderers:
https://github.com/vvvv/VL.Stride/blob/preview/2021.4/packages/VL.Stride.Runtime/src/Engine/SchedulerSystem.cs

to see that in action you can try it in vvvv, go into the help browser in the _Learn_ tab and look for _HowTo Render a Scene multiple times_.

![image](https://user-images.githubusercontent.com/1094716/151680395-ec52c525-fce1-4295-8e15-f1d934fd258b.png)

you can download it from: https://visualprogramming.net

what we don't have is rendering from multiple threads into multiple windows, this would be more tricky to achieve... it might be easier to just have multiple apps running and communicate over the network and shared textures.

**Reply by TheKeyblader** [reply]
Thx a lot this will help me.

I have some additionnal question about VL.Stride.

It's hardware accelerated ?
does it support all graphics api (OpenGL, Vulkan) ?
Inputs is rooted to stride ?

**Reply by tebjan** [reply]
Yes, it's hardware accelerated, it's all based on stride. We use dx11 as backed. We didn't look into other graphics backends yet, because we use a lot of compute shaders.
